consensus/src/chained_bft/consensus_types/block.rs

```diff
@@ -209,6 +242,43 @@ where
         )
     }
 
+    /// The NIL blocks are special: they're not carrying any real payload and are generated
+    /// independently by different validators just to fill in the round with some QC.
+    pub fn make_nil_block(parent_block: &Block<T>, round: Round, quorum_cert: QuorumCert) -> Self {
+        checked_precondition_eq!(quorum_cert.certified_block_id(), parent_block.id());
+        checked_precondition!(round > parent_block.round());
+
+        let payload = T::default();
+        // We want all the NIL blocks to agree on the timestamps even though they're generated
+        // independently by different validators, hence we're using the timestamp of a parent.
+        let timestamp_usecs = parent_block.timestamp_usecs;
+
+        let block_serializer = BlockSerializer {
+            parent_id: parent_block.id(),
+            payload: &payload,
+            round,
+            height: parent_block.height() + 1,
+            timestamp_usecs,
+            quorum_cert: &quorum_cert,
+            // the author here doesn't really matter for as long as all the NIL Blocks are hashing
+            // the same value, hence use the special genesis author for hashing.
+            author: default_author(),
```

Comment:
Is default_author needed?  Perhaps a way around it would be to pass the BlockSource here?